HousingFinanceInterimApi/V1/UseCase/LogParserUseCase.cs

```diff
@@ -0,0 +1,159 @@
+        {
+            if (!_logGroups.Any())
+            {
+                throw new ArgumentException("Log groups cannot be null or empty", nameof(_logGroups));
+            }
+
+            try
+            {
+                // Process log groups in parallel
+                var tasks = _logGroups.Select(async logGroup =>
+                {
+                    try
+                    {
+                        // Query CloudWatch Logs Insights
+                        var queryResults = await QueryCloudWatchLogs(logGroup).ConfigureAwait(false);
+
+                        // Evaluate results and update the database
+                        await _logParserGateway.UpdateDatabaseWithResults(logGroup, queryResults).ConfigureAwait(false);
+                    }
+                    catch (AmazonCloudWatchLogsException awsEx)
+                    {
+                        // Handle AWS-specific errors
+                        LoggingHandler.LogError($"AWS error for log group {logGroup}: {awsEx.Message}");
+
+                        await LogFailureToDatabase(logGroup, awsEx.Message).ConfigureAwait(false);
+                    }
+                    catch (DbUpdateException dbEx)
+                    {
+                        // Handle database-specific errors
+                        LoggingHandler.LogError($"Database update error for log group {logGroup}: {dbEx.Message}");
+                        throw;
+                    }
+                    catch (Exception ex)
+                    {
+                        // Handle other unexpected errors
+                        LoggingHandler.LogError($"Unexpected error for log group {logGroup}: {ex.Message}");
+
+                        await LogFailureToDatabase(logGroup, ex.Message).ConfigureAwait(false);
+                    }
+                });
+
+                // Wait for all tasks to complete
+                await Task.WhenAll(tasks).ConfigureAwait(false);
+
+                return new StepResponse() { Continue = true, NextStepTime = DateTime.Now.AddSeconds(int.Parse(_waitDuration)) };
+            }
+            catch (Exception ex)
+            {
+                LoggingHandler.LogError($"Error in Log Parser Lambda: {ex.Message}");
+                throw;
+            }
+        }
+
+        public async Task<List<List<ResultField>>> QueryCloudWatchLogs(string logGroupName)
+        {
+            try
+            {
+                // TODO:: Dev-Testing with RequestId for now - revert to "ERROR" later
+                var query = @"
+                        fields @timestamp, @message
+                        | filter @message like /RequestId/
+                        | sort @timestamp desc
+                        | limit 100";
+
+                var startQueryRequest = new StartQueryRequest
+                {
+                    LogGroupName = logGroupName,
+                    StartTime = new DateTimeOffset(DateTime.UtcNow.AddDays(-1)).ToUnixTimeMilliseconds(),
```

Comment:
Just a note:
Whichever is the monday-only procedure will return no results for most of the week.

Reply:
Yes - the log groups for the Housing Benefit lambda functions